flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/datastream/DataStreamUserDefinedFunctionITCase.scala

```diff
@@ -230,6 +231,26 @@ class DataStreamUserDefinedFunctionITCase extends StreamingMultipleProgramsTestB
     assertEquals(expected.sorted, StreamITCase.testResults.sorted)
   }
 
+
+  @Test
+  def testScalarFunctionDeterministic(): Unit = {
```

Comment:
I think an IT case can't cover this change. Whether the UDF is deterministic or not, the result should be same.